Tests/iaas/flavor-naming/flavor-add-extra-specs.py

```diff
@@ -0,0 +1,345 @@
+#!/usr/bin/env python3
+# vim: set ts=4 sw=4 et:
+"""
+flavor-add-extra-specs.py
+
+Cycles through all SCS- openstack flavors and adds properties specified in
+scs-0103-v1 <https://docs.scs.community/standards/scs-0103-v1-standard-flavors>.
+
+Usage: flavor-add-extra-specs.py [options] [FLAVORS]
+Options:
+    -h|--help:  Print usage information
+    -d|--debug: Output verbose debugging info
+    -q|--quiet: Don't output notes on changes performed
+    -t|--disk0-type TYPE:   Assumes disk TYPE for flavors w/ unspec disk0-type
+    -p|--cpu-type TYPE:     Assumes CPU TYPE for flavors w/o SCS name
+    -c|--os-cloud CLOUD:    Cloud to work on (default: OS_CLOUD env)
+    -n|--no-changes:        Do not perform any change
```

Comment:
Well, rather than performing the operations, you provide input for c'n'p for the changes.
This has some advantages:
- Operator can review the changes -- for understanding or because of lack of trust
- Operator could even add additional things if she wanted to
(Sorry, I don't see the elimination of an option with a global variable here as relevant. I know pylint gives me a lower rating ...)
It also has disadvantages:
- The tool can not just be called in order to accomplish a job. You force it to be interactive
- For lots of changes (several screens full) or for environments with limited c'n'p capabilities (think people that are not using Linux as their main desktop), this can be very inconvenient.

So, not convinced.

What I could imagine however doing: Reverse the default.
By default we don't perform the changes.
And with `-y|--yes-i-really-want-this` things would be performed. 

If you prefer the current `INFO` output to be stated as openstack flavor set ... commands, that should be feasible as well.

Reply:
Reversing the option is fine with me! Couldn't we generate the output and then perform it as well (using the shell)?

Reply:
Regarding the option: I do think we have an unhealthy proliferation of options with our tools. Operators usually don't want to read manuals. However, if the tool is used without the "do it" option, it could output a hint that this option exists.

Reply:
Thinking about collecting changes to only do one `openstack flavor set` call per flavor:
* We would probably remember the changes that need to happen in our own data structure with something like a verb (`DELETE`, `ADD`, `SET` or `DELETE`, `POST`, `PUT` if you prefer REST notation), an object and a property. So we'd end up with `[("DELETE", "scs:name-v3", None), ("POST", "scs:cpu-type", "shared-core"), ("PUT", "scs:name-v1", "SCS-1L:1"), ("POST", "scs:name-v2", "SCS-1L-1")]` for example.
* This could then be translated to an `openstack image set` call with the complexity, that there is no syntax to remove single properties, but just one to delete all, so in case there is any `DELETE` operation, we would have to re-add all existing properties, even those that do not change. This make it harder to review for the operator and removes some of the value of this feature ... So we'd probably output the list of changes for review rather than the OpenStack command ...
* The amount of API calls would be reduced, as we'd of course still use the list to consolidate calls. Unlike the CLI tool (which always takes an extra second to startup and get a token), doing API calls from the python SDK is not a performance issue, so this optimization does not seem important. 

I would think we leave this improvement to a v2 for the tool.
While it does make things more beautiful and clearer, it does also add some complexity on its own and I'm unsure whether it's worth it.

Reply:
> there is no syntax to remove single properties, but just one to delete all, so in case there is any DELETE operation, we would have to re-add all existing properties, even those that do not change

I couldn't believe it, so I had to have a look. And what I saw (cannot be unseen). It's such a horrible sh**show -- the whole range: python-openstackclient, openstacksdk, and not least nova, with its api, its objects and its db. This is so much boilerplate, so much bloat, and just so ... mediocre at best.

So we definitely won't do the translation to `python-openstack-client`. Screw that!

Reply:
So just one of the many peculiarities... there is an API endpoint for creating extra specs. But it is functionally equivalent with the update endpoint, which means:

- It won't fail if any extra specs are already present.
- It won't remove any keys at all.

Reply:
Yeah, it's a mess :-(